snapcraft/cli/_runner.py

```diff
@@ -64,6 +64,19 @@
 def run(ctx, debug, catch_exceptions=False, **kwargs):
     """Snapcraft is a delightful packaging tool."""
 
+    # Do this early, if we are in managed-host and not root yet we want to re-exec with
+    # sudo keeping the current environment settings.
+    if os.getenv("SNAPCRAFT_BUILD_ENVIRONMENT") == "managed-host" and os.geteuid() != 0:
+        snap_path = os.getenv("SNAP")
+        # Use -E to keep the environment.
+        cmd = ["usr/bin/sudo", "-E"]
+        # Pick the correct interpreter if running from a snap (in managed-host mode, this
+        # is most likely the only scenario).
+        if os.getenv("SNAP_NAME") == "snapcraft":
```

Comment:
Is it actually possible for this to _not_ be the case? Or should we error out as this point?

Reply:
testing is the case